scripts/dev/infra-sync.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# Fetches cluster artifacts from infra.rox.systems and merged them into a kubeconfig file.
+
+set -euo pipefail
+
+os=$(uname -o)
+infra_dir="${HOME}/.kube/infra"
+clusters_dir="${infra_dir}/clusters"
+infra_kubeconfig="${INFRA_KUBECONFIG:-${infra_dir}/kubeconfig}"
+extra_kubeconfigs="${EXTRA_KUBECONFIGS:-}"
+
+delete_cluster_artifacts() {
+    echo "Deleting old cluster artifacts..."
+    # Just a couple of safety checks to make sure that we are not attempting to delete
+    # something like `/`.
+    mkdir -p "${clusters_dir}"
+    if [[ ( "${#clusters_dir}" -le 1 ) || ( ! -d "${clusters_dir}" ) ]]; then
+        echo >&2 "The clusters directory '${clusters_dir}' does not seem properly initialized."
+        exit 1
+    fi
+    # A call of the form `rm -rf some_directory/*` would fail if `some_directory/*` doesn't
+    # evaluate to anything.
+    find "${clusters_dir}" -mindepth 1 -maxdepth 1 -type d -exec rm -rf {} +
+}
+
+export INFRA_TOKEN="${INFRA_TOKEN:-}"
+if [[ "${INFRA_TOKEN}" == "" ]]; then
+    case "${os}" in
+        "Darwin")
+            if token=$(security find-generic-password -s infra.rox.systems -D token -a "" -l acs -w); then
+                INFRA_TOKEN="${token}"
+            else
+                echo >&2 "Failed to retrieve token from Keychain"
+                echo >&2 "Use the following command for saving your infra token in Keychain:"
+                echo >&2
+                echo >&2 "  $ security add-generic-password -s infra.rox.systems -D token -a \"\" -l acs -T \"\" -w \"YOUR_INFRA_TOKEN\""
+                echo >&2
+                exit 1
+            fi
+            ;;
+        *)
+            echo >&2 "INFRA_TOKEN is not set and support for retrieving a token has not been implemented on ${os}."
+            exit 1
+            ;;
+    esac
+fi
+
+echo "Syncing your infra clusters..."
+delete_cluster_artifacts
+
+# Fetching artifacts for infra clusters.
```

Comment:
Speaking of readability: Would it make sense to also refactor the whole fetching of clusters into a function? 
Then, the main function of the script would look something more like
```
clean_infra_cluster_cache
fetch_infra_clusters
kubecfg_merge
```
which I find a lot more readable. There likely would be some error handling, but I think you get the idea 😄

Reply:
I did some refactoring along these lines, but I decided against doing the `clean_infra_cluster_cache` unconditionally at the beginning of the script, because I think it's a good idea to at least make sure that we were able to retrieve an infra token before beginning the deletion of existing artifacts... (therefore, doing it inside the `fetch_infra_clusters` function).

Reply:
ok to resolve @Maddosaurus?